### PR TITLE
rm: troubleshooting: add how to automatically load a kernel module

### PR DESCRIPTION
**Pull request by vanmaegima**
Signed-off-by: Vanessa Maegima <vanessa.maegima@foundries.io>

### CODE REVIEW

**Comment by doanac** [comment]
Docs for 36beba964ccde2328ab75942d2ae90a0b8eaf85d are browsable at: https://ci.foundries.io/projects/fio-docs/builds/1391/docs/artifacts/html/index.html

File: source/reference-manual/troubleshooting/troubleshooting.rst [hunk 1]
```diff
@@ -357,3 +357,48 @@ extended for the other net interfaces.
...
+.. code-block:: none
+
+    SUMMARY = "Module summary"
+    LICENSE = "GPLv2"
+    LIC_FILES_CHKSUM = "file://COPYING;md5=12f884d2ae1ff87c09e5b7ccc2c4ca7e"
+
+    inherit module
+
+    PR = "r1"
+    PV = "0.1"
```
Comment (angolini) [inline source/reference-manual/troubleshooting/troubleshooting.rst:389]: I would drop these 2 lines.

File: source/reference-manual/troubleshooting/troubleshooting.rst [hunk 2]
```diff
@@ -357,3 +357,48 @@ extended for the other net interfaces.
...
+
+**conf/machine/include/lmp-factory-custom.inc:**
+
+.. code-block:: none
+
+    KERNEL_MODULE_AUTOLOAD:<machine> = "i2c-dev"
+
+2. Adding a new driver/module to the Linux kernel source code:
+
+**meta-subscriber-overrides/recipes-kernel/kernel-modules/<module>.bb:**
```
Comment (angolini) [inline source/reference-manual/troubleshooting/troubleshooting.rst:378]:
```suggestion
**meta-subscriber-overrides/recipes-kernel/kernel-modules/<module>_<pv>.bb:**
```

I prefer to define PV in the recipe file name. It helps to manage when new revision is released (or to have more than one at a time)

File: source/reference-manual/troubleshooting/troubleshooting.rst [hunk 3]
```diff
@@ -357,3 +357,48 @@ extended for the other net interfaces.
...
+      file://COPYING \
+    "
+
+    S = "${WORKDIR}"
+
+    KERNEL_MODULE_AUTOLOAD:append = "<module>"
+
+Make sure to provide the source code and header for the new module, as well as
+the license and Makefile. Also make sure to adjust the provided values as
+needed by the recipe (``LICENSE``, ``PR``, ``PV``).
```
Comment (angolini) [inline source/reference-manual/troubleshooting/troubleshooting.rst:404]:
```suggestion
needed by the recipe (``LICENSE``, ``PV``).
```

**Review by angolini — APPROVED** [review]
LGTM, only minor detail

File: source/reference-manual/troubleshooting/troubleshooting.rst [hunk 4]
```diff
@@ -357,3 +357,48 @@ extended for the other net interfaces.
     method=auto
 
 Remember to adjust the `address1` parameter as needed.
+
+Automatically loading a kernel module
```
Comment (kprosise) [inline source/reference-manual/troubleshooting/troubleshooting.rst:361]:
```suggestion
Automatically Loading a Kernel Module
```

**Review by kprosise — APPROVED** [review]
LGTM

**Pull request merged**